### ISSUE TITLE
DateTimePicker not updating Value sometimes

### ISSUE THREAD

**Issue by FabianMaurerRedheads** [dotnet/winforms#1695]
Example Code:

```
Form formParent = new Form();
formParent.IsMdiContainer = true;
var formChild = new Form();
formChild.TopLevel = false;
var DateStart = new DateTimePicker();
DateStart.ValueChanged += (o, s) => MessageBox.Show("Success");
formChild.Controls.Add(DateStart);
formChild.Visible = true;
formChild.Parent = formParent;

Application.Run(formParent);
```

Try changing the date, there is no messagebox.

However, when exchanging the two lines
```
formChild.Visible = true;
formChild.Parent = formParent;
```
it suddenly works.

**Comment by weltkante** [comment]
There are a bunch of similar bugs where events aren't fired when initialization happens in a certain order, this was fixed in PR #1661 (preview 9), I'd guess this is probably related.

Unfortunately I don't know where to obtain preview 9, [nightly builds](https://github.com/dotnet/core/blob/master/daily-builds.md) seem to be already built against the 5.0 master branch even though the page and download link still advertise them as 3.0

**Comment by RussKie** [comment]
A Preview9 build has been just published on our internal feeds, so won't be too long before the build is available to the general public.

You can also pull and try release/3.0-preview9 branch to see if the issue is still repros.

**Comment by merriemcgaw** [comment]
@FabianMaurerRedheads is this behavior different in .NET Framework 4.5+? Or is this similar in both cases?

**Comment by FabianMaurerRedheads** [comment]
@merriemcgaw It always worked with .NET Framework, just Core was affected. However, starting with preview9 it works in Core as well. Thanks at all!

**Comment by weltkante** [comment]
Like I said above this was probably fixed by PR #1661 , since you confirmed it works I think it can be closed.
